.circleci/config.yml

```diff
@@ -0,0 +1,108 @@
+version: 2.1  # The version of CircleCI platform being used
+executors:
+  python-docker:
+    working_directory: ~/AutoDH 
+    docker:
+      - image: cimg/python:3.6
+
+
+jobs:  # The jobs level contains a collection of arbitrarily named children
+  run-tests:
+    executor: python-docker
+    steps:  # The steps collection is an ordered list of run directives
+      - checkout  # "checkout" checks out the source code to "working_directory"
+      - restore_cache:
+          key: pip-deps-{{ checksum "constraints.txt" }}-{{ checksum "setup.py" }}-{{ arch }}
+      - run:
+          name: Install Python Dependencies
+          command: |
+            set -euxo pipefail
+            python -m venv venv
+            source venv/bin/activate
+            pip install -U pip setuptools wheel
+            pip install -e '.[tests]' -c constraints.txt
+      - save_cache:
+          key: pip-deps-{{ checksum "constraints.txt" }}-{{ checksum "setup.py" }}-{{ arch }}
+          paths:
+            - venv
+      - run:
+          name: Run Linters
+          command: |
+            set -euxo pipefail
+            source venv/bin/activate
+            flake8 osaro_default_python_template tests
+            bandit -r --ini setup.cfg
+            black .
+            isort --recursive --check-only --diff --settings-path setup.cfg \
+              osaro_default_python_template tests
+            mypy --config-file setup.cfg
+      - run:
+          name: Run Pytest
+          command: |
+            set -euxo pipefail
+            source venv/bin/activate
+            pytest -c .circleci/resources/pytest_build_config.ini
+      - run:
+          name: Run Pytest With Mocking Disabled
+          command: |
+            set -euxo pipefail
+            source venv/bin/activate
+            export RUNTIME_ENVIRONMENT="build"
```

Comment:
what is happening here? why do we need to run the same test suite twice with different environment variables?

tests should generally accept parameters and not depend on the environment

Reply:
+1